src/object/authenticate.c

```diff
@@ -8656,20 +8656,20 @@ au_get_dba_user (void)
 }
 
 /*
- * au_check_serial_authorization - check whether the current user is able to
- *                                 modify serial object or not
+ * au_check_owner - check whether the current user is able to
+ *                                 modify row object or not
  *   return: NO_ERROR if available, otherwise error code
- *   serial_object(in): serial object pointer
+ *   record_object(in): row object pointer
  */
 int
-au_check_serial_authorization (MOP serial_object)
+au_check_owner (MOP record_object, int err_code)
```

Comment:
Instead of adding err_code parameter, how about call this function like the following pattern?
```
int
au_check_serial_authorization (MOP record_object)
{
  if (au_check_owner (record_object) != NO_ERROR)
  {
    return ER_QPROC_CANNOT_UPDATE_SERIAL;
  }
  else
  {
    return NO_ERROR;
  }
}
```

Reply:
For checking server authorization, the same pattern could be applied.

Reply:
If the return value of the au_check_owner() function is not NO_ERROR, and it is an error value that occurred in db_get(), it must be returned as it is.

Instead, if you add a wrapper function to handle it, it is possible without a parameter for the default error code.
Instead, the function call step will be incremented by one.
Even so, would it feel easy?

Reply:
Let's split the au_check_owner function into two.